### PR TITLE
UI/add event

### PR DESCRIPTION
**Pull request by Keith-Go**
## Ticket
- [Add event screen](https://trello.com/c/MzHuEXQN/23-add-event)

## To do
- Pick up image
- Show map
- Fill information event

## Video
https://github.com/luke-nguyen-goldenowl/join_me/assets/152830147/e49c5a50-1159-4bda-aa6d-2221a87d9b6a

### CODE REVIEW

**Comment by luke-nguyen-goldenowl** [comment]
hintText in the textFields in the Detail event tab should be blurred to make it easier to understand

File: lib/src/features/add_event/logic/add_event_state.dart [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:latlong2/latlong.dart';
+
+class AddEventState {
+  PageController controller;
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/add_event/logic/add_event_state.dart]: Don't put the controller in state, just put it in the Bloc class

File: lib/src/features/add_event/logic/add_event_state.dart [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  MapController mapController;
+  TextEditingController nameController;
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/add_event/logic/add_event_state.dart]: No need to use TextController, Just use String param
=> In UI/view: using XInput, the template already has, just use it and input the value and onChanged param to use

**Comment by Keith-Go** [comment]
resolved

File: lib/src/features/add_event/widget/address_page.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            const Text(
+              "Select location",
+              style: TextStyle(
+                fontSize: 18,
+                fontWeight: FontWeight.bold,
+              ),
+            ),
+            const SizedBox(height: 10),
+            Expanded(
+              child: FlutterMap(
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/add_event/widget/address_page.dart:28]: Update with Google Map Flutter later

**Comment by Keith-Go** [comment]
resolved

File: lib/src/features/add_event/widget/detail_page.dart [hunk 4]
```diff
@@ -23,166 +26,276 @@ class DetailPage extends StatelessWidget {
...
+                        lastDate: DateTime(2101),
+                      );
+                      if (selectedDate != null &&
+                          selectedDate != DateTime.now()) {
+                        addEventBloc.setDeadlineEvent(selectedDate);
+                      }
+                    },
+                  ),
+
+                  // TextFormField(
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/add_event/widget/detail_page.dart]: No need to comment code

**Comment by luke-nguyen-goldenowl** [comment]
After fixing, you can merge

**Pull request merged**